src/openforms/registrations/contrib/objects_api/migrations/0017_move_singleton_data.py

```diff
@@ -0,0 +1,89 @@
+    if not solo_model:
+        return
+
+    new_config_set = ObjectsAPIGroupConfig(
+        name="Objects APIs",
+        objects_service=solo_model.objects_service,
+        objecttypes_service=solo_model.objecttypes_service,
+        drc_service=solo_model.drc_service,
+        catalogi_service=solo_model.catalogi_service,
+        objecttype=solo_model.objecttype,
+        objecttype_version=solo_model.objecttype_version,
+        productaanvraag_type=solo_model.productaanvraag_type,
+        informatieobjecttype_submission_report=solo_model.informatieobjecttype_submission_report,
+        informatieobjecttype_submission_csv=solo_model.informatieobjecttype_submission_csv,
+        informatieobjecttype_attachment=solo_model.informatieobjecttype_attachment,
+        organisatie_rsin=solo_model.organisatie_rsin,
+        content_json=solo_model.content_json,
+        payment_status_update_json=solo_model.payment_status_update_json,
+    )
+
+    new_config_set.save()
+
+
+def reconfigure_objects_api_solo_model(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    ObjectsAPIConfig = apps.get_model("registrations_objects_api", "ObjectsAPIConfig")
+    ObjectsAPIGroupConfig = apps.get_model(
+        "registrations_objects_api", "ObjectsAPIGroupConfig"
+    )
+
+    solo_model = ObjectsAPIConfig.objects.first()
+    objects_api_group = ObjectsAPIGroupConfig.objects.all().order_by("pk").first()
+
+    if not solo_model or not objects_api_group:
+        return
+
+    solo_model.objects_service = objects_api_group.objects_service
+    solo_model.objecttypes_service = objects_api_group.objecttypes_service
+    solo_model.drc_service = objects_api_group.drc_service
+    solo_model.catalogi_service = objects_api_group.catalogi_service
+    solo_model.objecttype = objects_api_group.objecttype
+    solo_model.objecttype_version = objects_api_group.objecttype_version
+    solo_model.productaanvraag_type = objects_api_group.productaanvraag_type
+    solo_model.informatieobjecttype_submission_report = (
+        objects_api_group.informatieobjecttype_submission_report
+    )
+    solo_model.informatieobjecttype_submission_csv = (
+        objects_api_group.informatieobjecttype_submission_csv
+    )
+    solo_model.informatieobjecttype_attachment = (
+        objects_api_group.informatieobjecttype_attachment
+    )
+    solo_model.organisatie_rsin = objects_api_group.organisatie_rsin
+    solo_model.organisatie_rsin = objects_api_group.organisatie_rsin
+    solo_model.content_json = objects_api_group.content_json
+    solo_model.payment_status_update_json = objects_api_group.payment_status_update_json
+    solo_model.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("registrations_objects_api", "0016_objectsapigroupconfig"),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            move_singleton_data_to_objects_api_set_model,
+            reconfigure_objects_api_solo_model,
```

Comment:
I can live with a `migrations.RunPython.noop`, I make sure to make a database backup before checking out branches like this locally :)

The approach of taking the first group is also fine, but if the above remarks are taken into account, for the content_json defaults etc. you can just fall back to the model field defaults.

Reply:
Seems like you changed your mind from https://github.com/open-formulieren/open-forms/pull/3146/#discussion_r1228382591 :)

Reply:
ah, the growth of a person in a year's time :stuck_out_tongue: 

I don't have strong feelings about it - looking back I think the only ones actually doing rollbacks are developers, sometimes, when checking out branches locally to review/test things out. Making a DB dump and restoring that is probably less work than writing the reverse operation and writing tests for it :thinking:

Reply:
I did it as it was heavily inspired from Silvia's work on the ZGW groups, I'll see if I keep it when rewriting the migration history